catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -603,15 +606,20 @@ public boolean isPrivateField(String solrFieldName) {
    *     schema field name that matches the dynamic field type formatting.
    */
   public String getField(
-      String propertyName, AttributeFormat format, boolean isSearchedAsExactValue) {
+      String propertyName,
+      AttributeFormat format,
+      boolean isSearchedAsExactValue,
+      Map<String, Serializable> enabledFeatures) {
```

Comment:
❓ Is there a reason we don't pass `enabledFeatures` through the constructor of the `DynamicSchemaResolver` and make it an inner field? DynamicSchemaResolver seems to be the only one accessing this map and it would avoid having everyone passing it through as method arguments.

Reply:
The `DynamicSchemaResolver` gets created at the provider level and these features are applied at the individual query level.  If I construct the resolver with these features then they would apply to every query and I would still have to pass something into the method to determine whether I should turn it on/off.  At that point I felt it was easiest to just pass them into the function rather than tracking them as an instance variable and duplicating the data.